crates/l2/prover/zkvm/interface/guest/src/main.rs

```diff
@@ -16,26 +16,42 @@ fn main() {
     let mut state = EvmState::from(db.clone());
 
     // Validate the block pre-execution
-    validate_block(&block, &parent_block_header, &state).expect("invalid block");
+    if let Err(err) = validate_block(&block, &parent_block_header, &state) {
+        panic!("invalid block: {err}");
+    };
 
     // Validate the initial state
-    let (mut state_trie, mut storage_tries) = db
-        .build_tries()
-        .expect("failed to build state and storage tries or state is not valid");
+    let (mut state_trie, mut storage_tries);
+    match db.build_tries() {
+        Ok((state, storage)) => {
+            (state_trie, storage_tries) = (state, storage)
+        },
+        Err(err) => {
+            panic!("failed to build state and storage tries or state is not valid: {err}");
+        }
+    };
 
     let initial_state_hash = state_trie.hash_no_commit();
     if initial_state_hash != parent_block_header.state_root {
         panic!("invalid initial state trie");
     }
 
-    let receipts = execute_block(&block, &mut state).expect("failed to execute block");
-    validate_gas_used(&receipts, &block.header).expect("invalid gas used");
+    let receipts;
+    match execute_block(&block, &mut state) {
+        Ok(rec) => receipts = rec,
+        Err(err) => panic!("failed to execute block: {err}")
+    }
+
+    if let Err(err) = validate_gas_used(&receipts, &block.header) {
+        panic!("invalid gas used: {err}")
+    }
 
     let account_updates = get_state_transitions(&mut state);
 
     // Update tries and calculate final state root hash
-    update_tries(&mut state_trie, &mut storage_tries, &account_updates)
-        .expect("failed to update state and storage tries");
```

Comment:
ditto

Reply:
Fixed in 4e99d9ba79ec33aac1e0cd299a39ed2a9c2737e5